polars/polars-lazy/src/frame/mod.rs

```diff
@@ -870,20 +854,16 @@ impl LazyFrame {
         LazyGroupBy {
             logical_plan: self.logical_plan,
             opt_state,
-            keys: by.as_ref().to_vec(),
+            keys: by.into(),
             maintain_order: true,
             dynamic_options: Some(options),
             rolling_options: None,
         }
     }
 
     /// Similar to groupby, but order of the DataFrame is maintained.
-    pub fn groupby_stable<E: AsRef<[IE]>, IE: Into<Expr> + Clone>(self, by: E) -> LazyGroupBy {
-        let keys = by
-            .as_ref()
-            .iter()
-            .map(|e| e.clone().into())
-            .collect::<Vec<_>>();
+    pub fn groupby_stable<E: IntoIterator<Item = IE>, IE: Into<Expr>>(self, by: E) -> LazyGroupBy {
```

Comment:
this set of APIs is not aligned with the rest of them since they contain an `IE: Into<Expr>`. This change allows users to not have to create a slice to them be iterated by reference (and Expr cloned), instead allowing passing something that iterates by value